src/cache_cf.cc

```diff
@@ -2904,9 +2953,13 @@ dump_time_t(StoreEntry * entry, const char *name, time_t var)
 void
 parse_time_t(time_t * var)
 {
-    time_msec_t tval;
-    parseTimeLine(&tval, T_SECOND_STR, false);
-    *var = static_cast<time_t>(tval/1000);
+    static const auto maxTime = std::numeric_limits<time_t>::max();
+    const auto seconds = parseTimeLine<std::chrono::seconds>();
+    if (maxTime < seconds.count()) {
+        debugs(3, DBG_CRITICAL, "FATAL: time_t is too small to accommodate " << seconds.count() << " seconds");
```

Comment:
An admin may be confused by references to `time_t` (even though it is an official squid.conf directive "type"). I suggest rephrasing to avoid focusing on that implementation detail:
```suggestion
        debugs(3, DBG_CRITICAL, "FATAL: directive supports time values up to " << maxTime << " but given " << seconds.count() << " seconds");
```